### ISSUE TITLE
NTLM proxy authentication fails due to apache httpclient v5.0

### ISSUE THREAD

**Issue by DavidRipYmor** [bonigarcia/webdrivermanager#508]
**Description of the problem**: When downloading a webdriver (which usually works like a charm!) on a machine that uses NTLM to authenticate on the proxy, I get an apache error (see logs below). Naturally, I've double checked every proxy setting that was used, but it still kept throwing the same error. I've Googled this specific issue and it turns out to be a "minor" bug in the apache httpclient, which is resolved in the 5.0.1 release (https://stackoverflow.com/questions/61039771/apache-httpclient-v5-0-doesnt-authenticate-with-ntlm-on-windows). I've seen in your commits that you have upgraded this version already, however version 4.0.0 still uses httpclient v5.0. Could you create an update that includes the 5.0.1 version of the apache httpclient?

**Browser and version**: Internet Explorer 11.900.18362.0

**Operating system**: Windows 10, 64 bits

**WebDriverManager version**: 4.0.0

**WebDriverManager use**: 
```
public InternetExplorerBrowser(ProxyFactory proxyManager) throws IOException {
		proxyManager.getAsHttpProxy();

		if (!proxyManager.getProxyUsername().equals(proxyManager.getUndefined())) {
			WebDriverManager.iedriver().arch32().proxyUser(proxyManager.getProxyUsername());
			WebDriverManager.iedriver().arch32().proxyPass(proxyManager.getProxyPassword());
		}

		WebDriverManager.iedriver().arch32().proxy(proxyManager.getProxyServer() + ":" + proxyManager.getProxyPort())
				.setup();

		// Setup proxy
		setProxyManager(proxyManager);
	}
```

**WebDriverManager traces**: 
72 [main] INFO io.github.bonigarcia.wdm.WebDriverManager - Please answer the following questionnaire based on your experience with WebDriverManager. Thanks a lot!
77 [main] INFO io.github.bonigarcia.wdm.WebDriverManager - ====> http://tiny.cc/wdm-survey <====
1560 [main] INFO io.github.bonigarcia.wdm.WebDriverManager - Reading https://selenium-release.storage.googleapis.com/ to seek IEDriverServer
1684 [main] WARN org.apache.hc.client5.http.impl.classic.ConnectExec - Missing auth challenge
1684 [main] WARN org.apache.hc.client5.http.impl.classic.ConnectExec - Missing auth challenge
1700 [main] WARN org.apache.hc.client5.http.impl.classic.ConnectExec - Missing auth challenge
1709 [main] WARN io.github.bonigarcia.wdm.WebDriverManager - There was an error managing IEDriverServer (latest version) (org.apache.hc.client5.http.ClientProtocolException: CONNECT refused by proxy: HTTP/1.1 407 Proxy Authentication Required ( Forefront TMG requires authorization to fulfill the request. Access to the Web Proxy filter is denied.  )) ... trying again using latest driver stored in cache

**Error trace**:

**Comment by bonigarcia** [comment]
I plan to release a new version of WebDriverManager soon. Please stay tuned.

**Comment by bonigarcia** [comment]
I have just released WebDriverManager 4.1.0, it should be fixed in this version.
